src/client/Microsoft.Identity.Client/Platforms/Features/OpenTelemetry/OtelInstrumentation.cs

```diff
@@ -161,14 +163,36 @@ public void LogSuccessMetrics(
                 new(TelemetryConstants.CacheRefreshReason, authResultMetadata.CacheRefreshReason));
             }
 
-            if (s_durationInExtensionInMs.Value.Enabled && authResultMetadata.DurationCreatingExtendedTokenInUs > 0)
+            LogExtensionMetrics(authResultMetadata, platform, apiId);
+        }
+
+        private void LogExtensionMetrics(AuthenticationResultMetadata authenticationResultMetadata, string platform, ApiEvent.ApiIds apiId)
+        {
+            if (authenticationResultMetadata.DurationCreatingExtendedTokenInUs <= 0)
             {
-                s_durationInExtensionInMs.Value.Record(authResultMetadata.DurationCreatingExtendedTokenInUs,
-                new(TelemetryConstants.MsalVersion, MsalIdHelper.GetMsalVersion()),
-                new(TelemetryConstants.Platform, platform),
-                new(TelemetryConstants.ApiId, apiId),
-                new(TelemetryConstants.TokenSource, authResultMetadata.TokenSource),
-                new(TelemetryConstants.CacheLevel, authResultMetadata.CacheLevel));
+                return;
+            }
+
+            if (ExtensionCounters.Value.TryGetValue(authenticationResultMetadata.TelemetryTokenType, out Histogram<long> tokenCounter))
+            {
+                if (tokenCounter.Enabled)
+                {
+                    tokenCounter.Record(authenticationResultMetadata.DurationCreatingExtendedTokenInUs,
+                        new(TelemetryConstants.MsalVersion, MsalIdHelper.GetMsalVersion()),
+                        new(TelemetryConstants.Platform, platform),
+                        new(TelemetryConstants.ApiId, apiId),
+                        new(TelemetryConstants.TokenSource, authenticationResultMetadata.TokenSource),
+                        new(TelemetryConstants.CacheLevel, authenticationResultMetadata.CacheLevel));
+                }
+            }
+            else
+            {
+                Histogram<long> newHistogram = Meter.CreateHistogram<long>(
+                    $"MsalDuration{authenticationResultMetadata.TelemetryTokenType}Operation.1B",
```

Comment:
How many histograms are we expecting to create? I am expecting only with the TelemetryTokenType as Extention?

Reply:
Did you consider capturing token type as a tag instead? I think that should give you similar results instead of creating different histograms considering you are capturing same data in all of them.

Reply:
Reverting approach to capture tokenType instead